### PR TITLE
chore: capture BUILDKITE_RETRY_COUNT

### PR DESCRIPTION
**Pull request by estrada9166**
- Closes 

### Additional details

Capture Buildkite env variable [BUILDKITE_RETRY_COUNT](https://buildkite.com/docs/pipelines/environment-variables#BUILDKITE_RETRY_COUNT) - used to handle ci retries 

This env variable shows how many times this job has been retried - providing a value that would ensure that ci-build-id created (if it's not custom) is unique, making every retry a new run and avoiding having a passed run where the specs have been completed in the previous run (creating a false positive)

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by emilyrohrbough** [comment]
@estrada9166 Can you please update this PR summary to indicate why this env is needed and how it impacts the Cloud data?

File: cli/CHANGELOG.md
Comment (emilyrohrbough) [inline cli/CHANGELOG.md]:
```suggestion
- Capture the [Buildkite](https://buildkite.com/) CI provider's environment variable `BUILDKITE_RETRY_COUNT` to handle CI retries in the Cloud. Addressed in [25750](https://github.com/cypress-io/cypress/pull/25750).
```

**Review by lmiller1990 — APPROVED** [review]

**Comment by lmiller1990** [comment]
Looks good 👍 🪁

**Comment by emilyrohrbough** [comment]
@lmiller1990 This needed two approvals before merging. Also this didn't have a corresponding documentation updates....

**Pull request merged**
